kube/service.go

```diff
@@ -7,17 +7,50 @@ import (
 	"github.com/SUSE/fissile/model"
 	meta "k8s.io/client-go/pkg/api/unversioned"
 	apiv1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/runtime"
 	"k8s.io/client-go/pkg/util/intstr"
 )
 
-// NewClusterIPService creates a new k8s ClusterIP service
-func NewClusterIPService(role *model.Role, headless bool) (*apiv1.Service, error) {
-	if len(role.Run.ExposedPorts) == 0 {
-		// Kubernetes refuses to create services with no ports, so we should
-		// not return anything at all in this case
+// NewClusterIPServiceList creates a list of ClusterIP services
+func NewClusterIPServiceList(role *model.Role, headless bool) (*apiv1.List, error) {
+	list := &apiv1.List{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "List",
+		},
+		Items: []runtime.RawExtension{},
+	}
+	if headless {
+		svc, err := NewClusterIPService(role, true, false)
+		if err != nil {
+			return nil, err
+		}
+		if svc != nil {
+			list.Items = append(list.Items, runtime.RawExtension{Object: svc})
+		}
+	}
+	svc, err := NewClusterIPService(role, false, false)
```

Comment:
Because `true` / `false` arguments are opaque, we could do with a comment here (`// Create private services` or something).